pkg/util/rangeio/rangeio.go

```diff
@@ -377,17 +387,27 @@ func optimizeRanges(cfg *Config, in []Range) []Range {
 		i += 2 // Skip over the range we just inserted.
 	}
 
+	usedBuffers := []*bytes.Buffer{}
```

Comment:
nit: you could preallocate this to the target capacity (or length) since it'll have the same number of elements as out